code/datums/components/twohanded.dm

```diff
@@ -160,8 +191,18 @@
 	// wield update status
 	if(SEND_SIGNAL(parent, COMSIG_TWOHANDED_WIELD, user) & COMPONENT_TWOHANDED_BLOCK_WIELD)
 		return // blocked wield from item
+
+	//If wielder isn't null already, unreference the old wielder
+	if(wielder != null)
+		unreference_wielder()
+
+	wielder = user
 	wielded = TRUE
-	RegisterSignal(user, COMSIG_MOB_SWAP_HANDS, .proc/on_swap_hands)
+
+	RegisterSignal(user, COMSIG_PARENT_QDELETING, .proc/unreference_wielder)
```

Comment:
Actually call it on dropped

Reply:
Done, its called in unwield but it also needs to be done in QDELETING since the mob could be deleted while wielding the item